File: Code/Back-end/BEE-diamond-shop-system/BusinessObjects/Models/Bill.cs
using Domain.Constants;

namespace BusinessObjects.Models;

/// <summary>
/// Đơn bán hàng
/// </summary>
public partial class Bill
{
    public int BillId { get; set; }

    /// <summary>
    /// Khách hàng
    /// </summary>
    public int CustomerId { get; set; }

    /// <summary>
    /// Người bán
    /// </summary>
    public int UserId { get; set; }

    /// <summary>
    /// Quầy bán
    /// </summary>
    public int CounterId { get; set; }

    /// <summary>
    /// DiscountRate
    /// </summary>
    public decimal? DiscountRate { get; set; }

    /// <summary>
    /// Discount Description
    /// </summary>
    public string? DiscountDescription { get; set; }

    /// <summary>
    /// Tổng tiền
    /// </summary>
    public decimal? TotalAmount { get; set; }

    /// <summary>
    /// Ngày bán
    /// </summary>
    public DateTimeOffset SaleDate { get; set; }

    /// <summary>
    /// Loại hóa đơn (1: Bán hàng, 2: Mua hàng)
    /// </summary>
    public EnumBillType Type { get; set; } = EnumBillType.Sale;

    public DateTimeOffset CreatedAt { get; set; }
    public DateTimeOffset UpdatedAt { get; set; }

    public virtual ICollection<BillJewelry> BillJewelries { get; set; } = new List<BillJewelry>();
    public virtual ICollection<BillPromotion> BillPromotions { get; set; } = new List<BillPromotion>();
    public virtual ICollection<Warranty> Warranties { get; set; } = new List<Warranty>();
    public virtual ICollection<Transaction> Transactions { get; set; } = new List<Transaction>();
    public virtual Customer? Customer { get; set; }
    public virtual User? User { get; set; }
    public virtual Counter? Counter { get; set; }
}

File: Code/Front-end/fe_2/src/_mock/bill.js
export const bill = [
    {
      billId: 101,
      staffId: 1,
      customerId: 1001,
      totalAmount: 250.75,
      saleDate: '2023-01-15'
    },
    {
      staffId: 2,
      billId: 102,
      customerId: 1002,
      totalAmount: 175.50,
      saleDate: '2023-01-16'
    },
    {
      staffId: 3,
      billId: 103,
      customerId: 1003,
      totalAmount: 320.00,
      saleDate: '2023-01-17'
    },
    {
      staffId: 4,
      billId: 104,
      customerId: 1004,
      totalAmount: 450.00,
      saleDate: '2023-01-18'
    },
    {
      staffId: 1,
      billId: 105,
      customerId: 1005,
      totalAmount: 60.00,
      saleDate: '2023-01-19'
    },
    {
      staffId: 2,
      billId: 106,
      customerId: 1006,
      totalAmount: 980.25,
      saleDate: '2023-01-20'
    },
    {
      staffId: 3,
      billId: 107,
      customerId: 1007,
      totalAmount: 200.00,
      saleDate: '2023-01-21'
    },
    {
      staffId: 4,
      billId: 108,
      customerId: 1008,
      totalAmount: 150.75,
      saleDate: '2023-01-22'
    },
    {
      staffId: 1,
      billId: 109,
      customerId: 1009,
      totalAmount: 300.50,
      saleDate: '2023-01-23'
    },
    {
      staffId: 2,
      billId: 110,
      customerId: 1010,
      totalAmount: 400.00,
      saleDate: '2023-01-24'
    }
  ];

  export function addBill(newBillData) {
    bill.push(newBillData);
  } 
  